pkg/controller/backup/backup_controller.go

```diff
@@ -100,10 +105,84 @@ func (r *ReconcileBackup) Reconcile(request reconcile.Request) (reconcile.Result
 		if kerrors.IsNotFound(err) {
 			return reconcile.Result{}, nil
 		}
-
 		return reconcile.Result{}, err
 	}
 
+	if backup.ObjectMeta.DeletionTimestamp.IsZero() {
+		// The object is not being deleted, if it does not have this finalizer,
+		// add it and update the object.
+		if !slice.Contains(backup.ObjectMeta.Finalizers, Finalizer) {
+			log.Info("Adding finalizer")
+
+			backup.ObjectMeta.Finalizers = append(backup.ObjectMeta.Finalizers, Finalizer)
+			if err := r.Update(context.Background(), backup); err != nil {
+				return reconcile.Result{Requeue: true}, nil
+			}
+		}
+	} else {
+		// The object is being deleted, ensure that the finalizer exists then
+		// create a job to delete the restic snapshot.
+		if slice.Contains(backup.ObjectMeta.Finalizers, Finalizer) {
```

Comment:
So a finalizer is really just a flag on the object to do something in the reconcile loop?

Reply:
Yep exactly!